miner/worker.go

```diff
@@ -844,6 +860,47 @@ func (w *worker) commitTransactions(txs *types.TransactionsByPriceAndNonce, coin
 	return false
 }
 
+// commitNewTx is an OVM addition that mines a block with a single tx in it.
+// It needs to return an error in the case there is an error to prevent waiting
+// on reading from a channel that is written to when a new block is added to the
+// chain.
+func (w *worker) commitNewTx(interrupt *int32, tx *types.Transaction) error {
+	w.mu.RLock()
+	defer w.mu.RUnlock()
+	tstart := time.Now()
+
+	parent := w.chain.CurrentBlock()
+	// TODO: the timestamp is 0 until a l1 to l2 tx happens
+	timestamp := w.chain.CurrentTimestamp()
+
+	num := parent.Number()
+	header := &types.Header{
+		ParentHash: parent.Hash(),
+		Number:     num.Add(num, common.Big1),
+		GasLimit:   core.CalcGasLimit(parent, w.config.GasFloor, w.config.GasCeil),
+		Extra:      w.extra,
+		Time:       uint64(timestamp),
+	}
+	if err := w.engine.Prepare(w.chain, header); err != nil {
+		log.Error("Failed to prepare header for mining", "err", err)
+		return errors.New("Cannot mine block very bad")
+	}
+	// Could potentially happen if starting to mine in an odd state.
+	err := w.makeCurrent(parent, header)
+	if err != nil {
+		log.Error("Failed to create mining context", "err", err)
+		return errors.New("Cannot mine block very bad")
+	}
+	transactions := make(map[common.Address]types.Transactions)
+	acc, _ := types.Sender(w.current.signer, tx)
+	transactions[acc] = types.Transactions{tx}
```

Comment:
I don't really like having to create a `types.NewTransactionsByPriceAndNonce` here but it lets us use the same codepath as the rest of the miner. Was going for minimal diff

Reply:
What's the alternative here?

Reply:
The alternative is to extract the exact parts of `w.commit` and use them in this function here

Reply:
Since `w.commit` is used in the block creation process in the old codepath